### PR TITLE
Fixed a lot of issues with the create user api endpoint.

### PR DESCRIPTION
**Pull request by chisomdaniel**
The endpoint works fine now, the password is being hashed when you create a new user using the `curl`
commnd on the command line, but not when the admin dashbord is used and this is understandable
seeing that the code to hash the password occur at the api endpoint and
It seems like the admin dashboard isn't using this endpoint to create a new user.

Edit(Elisha): Yes the flask_admin uses a different method. Will have to interrogate it more.

### CODE REVIEW

**Comment by chisomdaniel** [comment]
ok they were not bugs actually, just some slight issues from my end. but I modified the code a little so you can merge this

File: back-end/api/rental_listing_api.py
Comment (elishambadi) [inline back-end/api/rental_listing_api.py:1]: Need to pass the args to the new RentalListing object in order to create a rental listing.
EDIT: Already seen it's use in user_api. Solid.

File: back-end/api/user_api.py [hunk 1]
```diff
@@ -54,19 +54,20 @@ def get(self):
...
+            parser.add_argument('profile_image', type=str, required=False, help='Image not provided')
             args = parser.parse_args()
-
             password_text = args['password']
         
             hashed_pw = bcrypt.hashpw(password_text.encode('utf-8'), bcrypt.gensalt())
+            args['password'] = hashed_pw
 
-            new_user = User(name=args['username'], email=args['email'], password=hashed_pw, type="tenant")
+            new_user = User(**args)
```
Comment (elishambadi) [inline back-end/api/user_api.py:70]: Well done to use **args. I had it in mind for the search_api. This is also great use for it!

File: back-end/api/rental_listing_api.py [hunk 2]
```diff
@@ -15,4 +15,23 @@ def get(self, listing_id=None):
...
+            parser = reqparse.RequestParser()
+            parser.add_argument('name', type=str, required=True, help='Rental name required')
+            parser.add_argument('location', type=str, required=True, help='Location required')
+            parser.add_argument('min-rent', type=float, required=True, help='Minimum rent required')
+            parser.add_argument('max-rent', type=float, required=True, help='Maximum rent name required')
+            parser.add_argument('rented', type=bool, required=True, help='Rent status required')
+            parser.add_argument('owner_id', type=int, required=True, help='Owner ID required')
+
+            args = parser.parse_args()
+            new_rental = RentalListing()
```
Comment (elishambadi) [inline back-end/api/rental_listing_api.py]: new_rental = RentalListing(**args)

File: back-end/api/rental_listing_api.py [hunk 3]
```diff
@@ -15,4 +15,23 @@ def get(self, listing_id=None):
         
         rentals = RentalListing.query.all()
         return jsonify([rental.serialize for rental in rentals])
```
Comment (elishambadi) [inline back-end/api/rental_listing_api.py]: rental.serialize(). Should be a method call not an attribute.

**Review by elishambadi — DISMISSED** [review]
A few changes to make.

File: back-end/api/rental_listing_api.py (hunk 2 above)
Comment (chisomdaniel) [reply back-end/api/rental_listing_api.py]: oh yes, thanks. I was still working on that api at the moment so it was intentional. I am done with it now tho

File: back-end/api/rental_listing_api.py (hunk 3 above)
Comment (chisomdaniel) [reply back-end/api/rental_listing_api.py]: thanks for this. it skipped me

Comment (elishambadi) [reply back-end/api/rental_listing_api.py]: Good to go.

File: back-end/api/rental_listing_api.py [hunk 4]
```diff
@@ -16,22 +19,78 @@ def get(self, listing_id=None):
...
+            parser.add_argument('owner_id', type=int, default=user_id, required=True, help='Owner ID required')
+            parser.add_argument('images', type=db.JSON, required=False, help='No image added')
+            parser.add_argument('latitude', type=float, required=False, help='Latitude not provided')
+            parser.add_argument('longitude', type=float, required=False, help='Longitude not provided')
 
             args = parser.parse_args()
-            new_rental = RentalListing()
+            new_rental = RentalListing(**args)
+            db.session.add(new_rental)
+            db.session.commit()
```
Comment (elishambadi) [inline back-end/api/rental_listing_api.py:42]: We should think about a db class to abstract the session.add and session.commit into one save() method.
But from next week maybe.

**Review by elishambadi — APPROVED** [review]
Looks robust.

**Pull request merged**